libs/langgraph/src/pregel/remote.ts

```diff
@@ -0,0 +1,446 @@
+  }
+
+  override async *_streamIterator(
+    input: PregelInputType,
+    options?: Partial<PregelOptions<Nn, Cc, ConfigurableFieldType>>
+  ): AsyncGenerator<PregelOutputType> {
+    const mergedConfig = mergeConfigs(this.config, options);
+    const sanitizedConfig = this._sanitizeConfig(mergedConfig);
+
+    const interruptBefore = this.interruptBefore ?? options?.interruptBefore;
+    const interruptAfter = this.interruptAfter ?? options?.interruptAfter;
+
+    yield* this.client.runs.stream(
+      sanitizedConfig.configurable.thread_id,
+      this.graphId,
+      {
+        input,
+        config: sanitizedConfig,
+        streamMode:
+          options?.streamMode !== undefined ? options.streamMode : "values",
+        interruptBefore: interruptBefore as string[],
+        interruptAfter: interruptAfter as string[],
+        streamSubgraphs: options?.subgraphs,
+      }
+    );
+  }
+
+  async updateState(
+    inputConfig: LangGraphRunnableConfig,
+    values: Record<string, unknown>,
+    asNode?: string
+  ): Promise<RunnableConfig> {
+    const mergedConfig = mergeConfigs(this.config, inputConfig);
+    const response = await this.client.threads.updateState(
+      mergedConfig.configurable?.thread_id,
+      { values, asNode, checkpoint: this._getCheckpoint(mergedConfig) }
+    );
+    return this._getConfig((response as any).checkpoint);
+  }
+
+  async *getStateHistory(
+    config: RunnableConfig,
+    options?: CheckpointListOptions
+  ): AsyncIterableIterator<StateSnapshot> {
+    const mergedConfig = mergeConfigs(this.config, config);
+    const states = await this.client.threads.getHistory(
+      mergedConfig.configurable?.thread_id,
+      {
+        limit: options?.limit ?? 10,
+        before: this._getCheckpoint(options?.before) as any,
+        metadata: options?.filter,
+        checkpoint: this._getCheckpoint(mergedConfig),
+      }
+    );
+    for (const state of states) {
+      yield this._createStateSnapshot(state);
+    }
+  }
+
+  protected _getDrawableNodes(
+    nodes: Array<{
+      id: string | number;
+      name?: string;
+      data?: Record<string, unknown>;
+      metadata?: unknown;
+    }>
+  ): Record<string, DrawableNode> {
+    const nodesMap: Record<string, DrawableNode> = {};
+    for (const node of nodes) {
+      const nodeId = node.id;
+      nodesMap[nodeId] = {
+        id: nodeId.toString(),
+        name: node.name ?? "",
+        data: (node.data as any) ?? {},
+        metadata: node.metadata as any,
+      };
+    }
+    return nodesMap;
+  }
+
+  async getState(
+    config: RunnableConfig,
+    options?: { subgraphs?: boolean }
+  ): Promise<StateSnapshot> {
+    const mergedConfig = mergeConfigs(this.config, config);
+
+    const state = await this.client.threads.getState(
+      mergedConfig.configurable?.thread_id,
+      {
+        checkpoint: this._getCheckpoint(mergedConfig),
+        subgraphs: options?.subgraphs,
+      } as any
+    );
```

Comment:
This doesn't seem to be right:

```suggestion
      this._getCheckpoint(mergedConfig),
      options
```

Reply:
Thanks, committing separately